wire-ios-data-model/Source/MLS/Migration/ProteusToMLSMigrationCoordinator.swift

```diff
@@ -148,6 +147,42 @@ public class ProteusToMLSMigrationCoordinator: ProteusToMLSMigrationCoordinating
         return .canStart
     }
 
+    /// This method is responsible for migrating all `mixed` group conversations to `mls`.
+    /// It evaluates each conversation to determine if the migration needs to be finalized (i.e: updating the protocol from `mixed` to `mls`)
+    /// or if it should first join the corresponding MLS group.
+    func migrateOrJoinGroupConversations() async {
+        do {
+            let conversations = try ZMConversation.fetchAllTeamGroupConversations(
+                messageProtocol: .mixed,
+                in: context
+            )
+
+            guard let mlsService = context.mlsService else {
```

Comment:
careful you're running on the swift concurrency pool (async method), accessing the mlsService needs to be done on the context thread (pool) 
```suggestion
            guard let mlsService = await context.perform { self.contex.mlsService } else {
```